### DISCUSSION TITLE
Home Assistant Community Store

### DISCUSSION THREAD

**Discussion by Curt4U** [eet-energy/solmate-sdk#12] [General]
wie wäre es mit eine integration dort?
what about an integration in hass?

**Comment by mmattel** [comment]
Maybe you want to have a look at: https://github.com/mmattel/EET-Solmate

**Comment by georgkapellerEET** [comment]
Hey!

We are actively working on providing an easy-to-use REST interface for our SolMate, it will be released still this year.
With this interface it should also be very easy to integrate SolMate into HASS.

Can you elaborate on your specific use cases so we can better understand how exactly you would want to use the battery there?

Until our release of the new interface, @mmattel (link above) has done a lot of good work to integrate the SolMate into HASS even with the current interfaces (websockets), so pls check out his project.

**Comment by Curt4U** [comment]
hi, just for information.. i am new to HASS, I am using fibers system for my home integration... both systems can install plugins/addons... I am looking to find a way to integrate your solar system to this home system. for the moment. I am using a power adapter (zig-bee & zwave) to get the power information. but it is very complicated for fiber... hard to find s way to define it es power generator... FIBARO Home Center 3 Energy Panel explained! docs.joepverhaeg.nl Solar Energiemonitor / Messen forum.siio.de on hass it is easier to select the device for solar production. so it can look like this https://community-assets.home-assistant.io/original/4X/8/7/4/874cf22023a958951a6d3b65f01c861646cbcceb.jpeg thx kurt Georg Kapeller ***@***.***> schrieb am 25. Sept. 2024 um 10:23: Hey! We are actively working on providing an easy-to-use REST interface for our SolMate, it will be released still this year. With this interface it should also be very easy to integrate SolMate into HASS. Can you elaborate on your specific use cases so we can better understand how exactly you would want to use the battery there? Until our release of the new interface, @mmattel (link above) has done a lot of good work to integrate the SolMate into HASS even with the current interfaces (websockets), so pls check out his project. — Reply to this email directly, view it on GitHub , or unsubscribe . You are receiving this because you authored the thread. Message ID: ***@***.***>

**Reply by mmattel** [reply]
Have a look at my repo. You can run this also as plain Python. The only thing it needs is a MQTT broker to connect. If you are not using HA, you must configure your target systen for the MQTT entities provided. HA has autoconfigure for this. This should not be a big problem when using MQTT explorer to identify the configs.

Note, getting the data into your home automation is a different task than displayin values in the way you want...

**Comment by Fox7AT** [comment]
Hi, I have limited IT knowledge. Together with Chat-GPT I was able to create a python program to read the values from my solmate and write the (min=max) injection_power based on the current energy price (awattar).

Unfortunately I have still the issue with droped connections from time to time, which means a manual restart of the scripts. In this periode I am missing data :( .
As I have currently one year of data on my sensors, I'm not sure how easy it is to change to @mmattel approach without loosing all my historic data.

Is it planed to create an addon for the HACS? Mayby together with @mmattel ?
I'd very much like to have an easy to use HACS addon.

**Reply by mmattel** [reply]
Having an open word, if EET does not mind...

Many thanks for the nice words about my software named [esham](https://github.com/mmattel/EET-Solmate). From the users I know and what I can prove myself, we have 100% uptime. I even dont think about that it is running. It is an absolutely no-brainer post installing and survives whatever I, my HW or the Solmate does. If you have fiddeld around with chat-GPT creating a Python program, it is easier to install esham via AppDaemon as described in my repo and avoid all the hassles and frustration you are facing right now.

With the program you created, to which target SW you are going to use, you will most likely lose all the historic data due to the fact how this data is stored in HA. You need to have some in depth knowledge to keep it. I did a closer look into and I can tell it is quite complicated (hacking the DB etc)... Advice, forget it. Make a hard cut and turn to a working solution.

Finally, the only and imho long term solution to integrate Solmates into HA is either using esham or when EET integrates its functionaity into the Solmate. Then, you only need a connection to a MQTT broker and HA interlinks with the broker. esham and any other SW is just a mediator between a Solmate and a MQTT broker, nothing more, but also not less :smiley:

**Reply by Fox7AT** [reply]
Thank you @mmattel for your response.
